From 6dbf088b5f5978fd554a71baf6d84777eab27984 Mon Sep 17 00:00:00 2001
From: Fabian Thomas <fabian@fabianthomas.de>
Date: Sat, 11 Feb 2023 00:21:27 +0100
Subject: 0-vim-like-bindings.patch

---
 shell/ev-application.c | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/shell/ev-application.c b/shell/ev-application.c
index ac60e47f..f19b758c 100644
--- a/shell/ev-application.c
+++ b/shell/ev-application.c
@@ -813,26 +813,26 @@ static void
 ev_application_startup (GApplication *gapplication)
 {
         const gchar *action_accels[] = {
-          "win.open",                   "<Ctrl>O", NULL,
-          "win.open-copy",              "<Ctrl>N", NULL,
+          "win.open",                   "<Ctrl><shift>O", NULL,
+          "win.open-copy",              "<Ctrl><shift>N", NULL,
           "win.save-as",                "<Ctrl>S", NULL,
-          "win.print",                  "<Ctrl>P", NULL,
+          "win.print",                  "<Ctrl><shift>P", NULL,
           "win.show-properties",        "<alt>Return", NULL,
           "win.copy",                   "<Ctrl>C", "<Ctrl>Insert", NULL,
           "win.select-all",             "<Ctrl>A", NULL,
-          "win.add-bookmark",           "<Ctrl>D", NULL,
-          "win.delete-bookmark",        "<Ctrl><Shift>D", NULL,
+          /* "win.add-bookmark",           "<Ctrl>D", NULL, */
+          /* "win.delete-bookmark",        "<Ctrl><Shift>D", NULL, */
           "win.close",                  "<Ctrl>W", NULL,
           "win.escape",                 "Escape", NULL,
           "win.find",                   "<Ctrl>F", "slash", "KP_Divide", NULL,
-          "win.find-next",              "<Ctrl>G", "F3", NULL,
-          "win.find-previous",          "<Ctrl><Shift>G", "<Shift>F3", NULL,
+          "win.find-next",              "<Ctrl>G", "F3", "<Ctrl>n", NULL,
+          "win.find-previous",          "<Ctrl><Shift>G", "<Shift>F3", "<Ctrl>p", NULL,
           "win.select-page",            "<Ctrl>L", NULL,
           "win.go-backwards",           "<Shift>Page_Up", NULL,
           "win.go-forward",             "<Shift>Page_Down", NULL,
-          "win.go-next-page",           "n", "<Ctrl>Page_Down", NULL,
-          "win.go-previous-page",       "p", "<Ctrl>Page_Up", NULL,
-          "win.go-back-history",        "<alt>P", "Back", NULL,
+          "win.go-next-page",           "n", "<Ctrl>Page_Down", "<shift>j", "<ctrl>d", NULL,
+          "win.go-previous-page",       "p", "<Ctrl>Page_Up", "<shift>k", "<ctrl>u", NULL,
+          "win.go-back-history",        "<alt>P", "Back", "<ctrl>o", NULL,
           "win.go-forward-history",     "<alt>N", "Forward", NULL,
           "win.sizing-mode::fit-page",  "f", NULL,
           "win.sizing-mode::fit-width", "w", NULL,
@@ -842,7 +842,8 @@ ev_application_startup (GApplication *gapplication)
           "win.caret-navigation",       "F7", NULL,
           "win.zoom-in",                "plus", "<Ctrl>plus", "KP_Add", "<Ctrl>KP_Add", "equal", "<Ctrl>equal", NULL,
           "win.zoom-out",               "minus", "<Ctrl>minus", "KP_Subtract", "<Ctrl>KP_Subtract", NULL,
-          "win.show-side-pane",         "F9", NULL,
+          // TODO: fix alt+s keybinding, probably problem with toolbar (check xournalpp alt+s)
+          "win.show-side-pane",         "<alt><shift>s", "F9", NULL,
           "win.fullscreen",             "F11", NULL,
           "win.presentation",           "F5", "<Shift>F5", NULL,
           "win.continuous",             "c", NULL,
-- 
2.44.0

