From ccc8cd2d460fdddaeee171004209d4a44a60eebb Mon Sep 17 00:00:00 2001
From: Fabian Thomas <fabian@fabianthomas.de>
Date: Fri, 10 Feb 2023 23:42:59 +0100
Subject: add-keyboard-binding-to-toggle-menu-bar.patch

---
 shell/ev-application.c |  1 +
 shell/ev-window.c      | 13 +++++++++++++
 2 files changed, 14 insertions(+)

diff --git a/shell/ev-application.c b/shell/ev-application.c
index f19b758c..39dfe1f7 100644
--- a/shell/ev-application.c
+++ b/shell/ev-application.c
@@ -839,6 +839,7 @@ ev_application_startup (GApplication *gapplication)
           "win.sizing-mode::automatic", "a", NULL,
           "win.default-zoom",           "<Ctrl>0", "<Ctrl>KP_0", NULL,
           "win.toggle-menu",            "F10", NULL,
+          "win.toggle-menubar",         "<alt>m", "F12", NULL,
           "win.caret-navigation",       "F7", NULL,
           "win.zoom-in",                "plus", "<Ctrl>plus", "KP_Add", "<Ctrl>KP_Add", "equal", "<Ctrl>equal", NULL,
           "win.zoom-out",               "minus", "<Ctrl>minus", "KP_Subtract", "<Ctrl>KP_Subtract", NULL,
diff --git a/shell/ev-window.c b/shell/ev-window.c
index 473933f9..2ed77763 100644
--- a/shell/ev-window.c
+++ b/shell/ev-window.c
@@ -642,6 +642,18 @@ ev_window_update_actions_sensitivity (EvWindow *ev_window)
 				      !recent_view_mode);
 }
 
+static void
+ev_window_cmd_action_menubar (GSimpleAction *action,
+			   GVariant      *parameter,
+			   gpointer       user_data)
+{
+	EvWindow *ev_window = user_data;
+	EvWindowPrivate *priv = GET_PRIVATE (ev_window);
+
+	GtkWidget* menubar_widget = GTK_WIDGET(ev_toolbar_get_header_bar (EV_TOOLBAR (priv->toolbar)));
+	gtk_widget_set_visible(menubar_widget, !gtk_widget_get_visible(menubar_widget));
+}
+
 static void
 set_widget_visibility (GtkWidget *widget, gboolean visible)
 {
@@ -6325,6 +6337,7 @@ static const GActionEntry actions[] = {
 	{ "default-zoom", ev_window_cmd_set_default_zoom },
 	{ "escape", ev_window_cmd_escape },
 	{ "toggle-menu", ev_window_cmd_action_menu },
+	{ "toggle-menubar", ev_window_cmd_action_menubar },
 	{ "caret-navigation", NULL, NULL, "false", ev_window_cmd_view_toggle_caret_navigation },
 	{ "add-annotation", ev_window_cmd_add_annotation },
 	{ "highlight-annotation", ev_window_cmd_add_highlight_annotation },
-- 
2.44.0

